src/Billing/Controllers/StripeController.cs

```diff
@@ -1,36 +1,133 @@
+        IOrganizationRepository organizationRepository,
+        ITransactionRepository transactionRepository,
+        IUserService userService,
+        IAppleIapService appleIapService,
+        IMailService mailService,
+        IReferenceEventService referenceEventService,
+        ILogger<StripeController> logger,
+        ITaxRateRepository taxRateRepository,
+        IUserRepository userRepository,
+        ICurrentContext currentContext,
+        IStripeEventService stripeEventService,
+        IStripeFacade stripeFacade,
+        StripeWebhookHandler webhookHandler)
+    {
+        _billingSettings = billingSettings?.Value;
+        _hostingEnvironment = hostingEnvironment;
+        _organizationService = organizationService;
+        _validateSponsorshipCommand = validateSponsorshipCommand;
+        _organizationSponsorshipRenewCommand = organizationSponsorshipRenewCommand;
+        _organizationRepository = organizationRepository;
+        _transactionRepository = transactionRepository;
+        _userService = userService;
+        _appleIapService = appleIapService;
+        _mailService = mailService;
+        _referenceEventService = referenceEventService;
+        _taxRateRepository = taxRateRepository;
+        _userRepository = userRepository;
+        _logger = logger;
+        _btGateway = new BraintreeGateway
+        {
+            Environment = globalSettings.Braintree.Production ?
+                Braintree.Environment.PRODUCTION : Braintree.Environment.SANDBOX,
+            MerchantId = globalSettings.Braintree.MerchantId,
+            PublicKey = globalSettings.Braintree.PublicKey,
+            PrivateKey = globalSettings.Braintree.PrivateKey
+        };
+        _currentContext = currentContext;
+        _globalSettings = globalSettings;
+        _stripeEventService = stripeEventService;
+        _stripeFacade = stripeFacade;
+        _webhookHandler = webhookHandler;
+
+        // Set up the chain of responsibility
+        var subscriptionDeletedHandler = new SubscriptionDeletedHandler(_organizationService, _userService, _stripeEventService);
+        var subscriptionUpdatedHandler = new SubscriptionUpdatedHandler(_organizationService, _userService,
+            _stripeEventService, _organizationSponsorshipRenewCommand);
+        var upcomingInvoiceHandler = new UpcomingInvoiceHandler(_organizationRepository, _userService,
+            _stripeEventService, _stripeFacade, _logger, _taxRateRepository, _validateSponsorshipCommand, _mailService);
+        var chargeSucceededHandler = new ChargeSucceededHandler(_transactionRepository, _logger, _stripeEventService);
+
+        subscriptionDeletedHandler.SetNextHandler(subscriptionUpdatedHandler);
+        subscriptionUpdatedHandler.SetNextHandler(upcomingInvoiceHandler);
+        upcomingInvoiceHandler.SetNextHandler(chargeSucceededHandler);
+
+        _webhookHandler = subscriptionDeletedHandler;
```

Comment:
I'm a bit confused as to what's going on here. Is this essentially to bypass some sort of routing mechanism? To me, it seems like it might be simpler to have this routing logic centrally located in some sort of `Handler Resolver`. If every handler implements the same interface, for instance, `IStripeWebhookHandler`, then you could easily spin up a service that pivots on the webhook type and initializes the required handler:

```csharp
return event.Type switch
{
    "invoice.upcoming" => new InvoiceUpcomingHandler(_dependency1, _dependency2),
    ...
}
```